### PR TITLE
Fix get segment_id by name for infographics by location

### LINKED ISSUE #2077: Fix get_road_segment_by_name()[Bug]
Fix get infographics by location when `road_segment_name` contains a junction that contains a dash (`-`).

### PR DESCRIPTION
**Pull request by ziv17**
Fixes #2077

### CODE REVIEW

File: anyway/parsers/location_extraction.py [hunk 1]
```diff
@@ -51,6 +51,22 @@ def get_road_segment_by_name(road_segment_name: str) -> RoadSegments:
     return segment
 
 
+def get_road_segment_by_name_and_road(road_segment_name: str, road: int) -> RoadSegments:
+    try:
+        from anyway.app_and_db import db
+    except ModuleNotFoundError:
+        pass
```
Comment (BarVolunteering) [inline anyway/parsers/location_extraction.py:58]: In this case should we log & raise an error?

**Review by BarVolunteering — COMMENTED** [review]
Looks good!
One question below

**Review by atalyaalon — APPROVED** [review]
looks good - I approve.
What about adding the road_segment_id in News Flash table and performing backfill? Do you think it should be instead / in addition?

**Comment by ziv17** [comment]
> What about adding the road_segment_id in News Flash table and performing backfill? Do you think it should be instead / in addition?

Hi @atalyaalon 
I think it is a good idea. I would add `segment_id` without removing `segment_name`. I understand it can be done as an `alembic` version, right?

**Pull request merged**